### PR TITLE
[FLINK-2008][FLINK-2296] Fix checkpoint committing & KafkaITCase [wip]

### PR DESCRIPTION
**Pull request by rmetzger**

### CODE REVIEW

File: docs/apis/streaming_guide.md [hunk 1]
```diff
@@ -1300,6 +1300,9 @@ Another way of exposing user defined operator state for the Flink runtime for ch
 
 When the user defined function implements the `Checkpointed` interface, the `snapshotState(…)` and `restoreState(…)` methods will be executed to draw and restore function state.
 
+In addition to that, user functions can also implement the `CheckpointNotifier` interface to receive notifications on completed checkpoints via the `notifyCheckpointComplete(long checkpointId)` method.
+Note that there is no guarantee for the user function to receive a notification once the checkpoint is complete.
```
Comment (StephanEwen) [inline docs/apis/streaming_guide.md]: Let's write it like "Note that there is no guarantee for the user function to receive a notification if a failure happens between checkpoint completion and notification. The notifications should hence be treated in a way that notifications from later checkpoints can subsume missing notifications."

File: flink-tests/src/test/resources/log4j-test.properties [hunk 2]
```diff
@@ -18,7 +18,7 @@
 
 # Set root logger level to OFF to not flood build logs
 # set manually to INFO for debugging purposes
-log4j.rootLogger=OFF, testlogger
+log4j.rootLogger=INFO, testlogger
```
Comment (StephanEwen) [inline flink-tests/src/test/resources/log4j-test.properties]: was this accidentally committed?

File: flink-tests/src/test/java/org/apache/flink/test/checkpointing/StreamCheckpointingITCase.java [hunk 3]
```diff
@@ -53,6 +62,8 @@
 @SuppressWarnings("serial")
 public class StreamCheckpointingITCase {
 
+	private static final Logger LOG = LoggerFactory.getLogger(StreamCheckpointingITCase.class);
```
Comment (StephanEwen) [inline flink-tests/src/test/java/org/apache/flink/test/checkpointing/StreamCheckpointingITCase.java]: These changes here also seem accidentally committed

File: flink-tests/src/test/resources/log4j-test.properties (hunk 2 above)
Comment (rmetzger) [inline flink-tests/src/test/resources/log4j-test.properties]: Yes

File: flink-tests/src/test/java/org/apache/flink/test/checkpointing/StreamCheckpointingITCase.java (hunk 3 above)
Comment (rmetzger) [inline flink-tests/src/test/java/org/apache/flink/test/checkpointing/StreamCheckpointingITCase.java]: Before I decided to undo parts of the checkpointing changes (sending the full state back  from the JM to the tasks), I had added an additional test there.

I can delete the logger factory.

File: flink-staging/flink-streaming/flink-streaming-connectors/flink-connector-kafka/src/main/java/org/apache/flink/streaming/connectors/kafka/Utils.java [hunk 4]
```diff
@@ -53,13 +62,22 @@ public boolean isEndOfStream(T nextElement) {
 
 		@Override
 		public byte[] serialize(T element) {
-			DataOutputSerializer dos = new DataOutputSerializer(16);
+			if(dos == null) {
+				dos = new DataOutputSerializer(1);
```
Comment (StephanEwen) [inline flink-staging/flink-streaming/flink-streaming-connectors/flink-connector-kafka/src/main/java/org/apache/flink/streaming/connectors/kafka/Utils.java]: I think a starting size of `16` is nicer than one of `1`

**Comment by StephanEwen** [comment]
Except for the issue mentioned by Gyula (the double commit of the head), this looks good. I would like to merge this later today or tomorrow. Could address Gyula's comment while merging...

**Comment by gyfora** [comment]
:+1:

**Pull request closed without merging**